parsing/unit_info.mli

```diff
@@ -58,15 +58,27 @@ type t
     - the module name associated to the unit
     - the filename prefix (dirname + basename with all extensions stripped)
       for compilation artifacts
-    - the input source file
+    - the original input source file
+    - the raw input source file
     For instance, when calling [ocamlopt dir/x.mli -o target/y.cmi],
     - the input source file is [dir/x.mli]
     - the module name is [Y]
     - the prefix is [target/y]
+
+    When calling, for example, [ocamlopt foo.pp.ml] (where foo.pp.ml is a serialized,
+    ppx-expanded AST of foo.ml), the "raw" source file is foo.pp.ml, while foo.ml is the
+    "original" source file.
 *)
 
-(** [source_file u] is the source file of [u]. *)
-val source_file: t -> filename
+(** [original_source_file u] is the original source file of [u]. When calling,
+    for example, [ocamlopt foo.pp.ml] (where foo.pp.ml is a serialized, ppx-expanded AST
+    of foo.ml), foo.ml is the "original" source file. *)
+val original_source_file: t -> filename
+
+(** [raw_source_file u] is the raw source file of [u]. When calling, for example,
+    [ocamlopt foo.pp.ml] (where foo.pp.ml is a serialized, ppx-expanded AST of foo.ml),
+    foo.pp.ml is the "raw" source file.) *)
+val raw_source_file: t -> filename
```

Comment:
After a quick glance, I have the impression that this `raw_source_file` field is not worth the added complexity. The source file that matters is the human-readable one. Typically, the compiler profiling information should also use the human-readable source file.